tests/fixture/testnet/local/network.go

```diff
@@ -0,0 +1,658 @@
+		dataDir = filepath.Join(ln.Dir, node.NodeID.String())
+		flags[cfg.DataDirKey] = dataDir
+	}
+
+	return nil
+}
+
+// Starts a network for the first time
+func (ln *LocalNetwork) Start(w io.Writer) error {
+	if len(ln.Dir) == 0 {
+		return errors.New("local network directory not set - has Create() been called?")
+	}
+
+	// Ensure configuration on disk is current
+	if err := ln.WriteAll(); err != nil {
+		return err
+	}
+
+	// Accumulate bootstrap nodes such that each subsequently started
+	// node bootstraps from the nodes previously started.
+	//
+	// e.g.
+	// 1st node: no bootstrap nodes
+	// 2nd node: 1st node
+	// 3rd node: 1st and 2nd nodes
+	// ...
+	//
+	bootstrapIDs := []string{}
+	bootstrapIPs := []string{}
+
+	// Configure networking and start each node
+	for i, node := range ln.Nodes {
+		// Update network configuration and write to disk
+		httpPort := 0
+		stakingPort := 0
+		if ln.UseStaticPorts {
+			httpPort = int(ln.InitialStaticPort) + i*2
+			stakingPort = httpPort + 1
+		}
+		node.SetNetworkingConfig(httpPort, stakingPort, bootstrapIDs, bootstrapIPs)
+
+		// Write configuration to disk in preparation for node start
+		if err := node.WriteConfig(); err != nil {
+			return err
+		}
+
+		if err := node.Start(w, ln.ExecPath); err != nil {
+			return err
+		}
+
+		bootstrapIP := ""
+		if ln.UseStaticPorts {
+			// Infer the bootstrap IP from flag configuration
+			rawStakingHost, ok := node.Flags[cfg.StakingHostKey]
+			// Only use the default value if the key is not set to allow
+			// an empty value (i.e. bind all interfaces) to be supplied.
+			stakingHost := "127.0.0.1"
+			if ok {
+				stakingHost = cast.ToString(rawStakingHost)
+			}
+			stakingPort := strconv.Itoa(cast.ToInt(node.Flags[cfg.StakingPortKey]))
+			bootstrapIP = net.JoinHostPort(stakingHost, stakingPort)
+		} else {
+			// Use the bootstrap ip from process context
+			if err := node.WaitForProcessContext(context.TODO()); err != nil {
+				return err
+			}
+			bootstrapIP = node.StakingAddress
+		}
+		bootstrapIDs = append(bootstrapIDs, node.NodeID.String())
+		bootstrapIPs = append(bootstrapIPs, bootstrapIP)
+	}
+
+	return nil
+}
+
+// Wait until all nodes in the network are healthy.
+func (ln *LocalNetwork) WaitForHealth(ctx context.Context, w io.Writer) error {
+	healthyNodes := set.NewSet[ids.NodeID](len(ln.Nodes))
+	for {
+		for _, node := range ln.Nodes {
+			select {
+			case <-ctx.Done():
+				return errors.New("failed to see healthy nodes before timeout")
+			default:
+				if healthyNodes.Contains(node.NodeID) {
+					continue
+				}
+				healthy, err := node.IsHealthy(ctx)
+				if err != nil {
+					return err
+				}
+				if healthy {
+					healthyNodes.Add(node.NodeID)
+					if _, err := fmt.Fprintf(w, "%s is healthy @ %s\n", node.NodeID, node.URI); err != nil {
+						return err
+					}
+				}
+			}
+		}
+		if len(healthyNodes) == len(ln.Nodes) {
+			break
+		}
+		time.Sleep(networkHealthCheckInterval)
+	}
+	return nil
```

Comment:
I think there are generally better options than `time.Sleep`.
```suggestion
	ticker := time.NewTicker(networkHealthCheckInterval)
	defer ticker.Stop()

	healthyNodes := set.NewSet[ids.NodeID](len(ln.Nodes))
	for healthyNodes.Len() < len(ln.Nodes) {
		for _, node := range ln.Nodes {
			if healthyNodes.Contains(node.NodeID) {
				continue
			}

			healthy, err := node.IsHealthy(ctx)
			if err != nil {
				return err
			}

			if healthy {
				healthyNodes.Add(node.NodeID)
				if _, err := fmt.Fprintf(w, "%s is healthy @ %s\n", node.NodeID, node.URI); err != nil {
					return err
				}
			}
		}

		select {
		case <-ctx.Done():
			return errors.New("failed to see healthy nodes before timeout")
		case <-ticker.C:
		}
	}
	return nil
```